src/irmin-pack/unix/control_file_intf.ml

```diff
@@ -196,6 +196,10 @@ module type S = sig
       {3 RO mode}
 
       Always returns [Error `Ro_not_allowed]. *)
+
+  val gc_generation : t -> (int, [> `Gc_disallowed ]) result
+  (** Returns the current generation, or returns an error if there is no
+      generation in the control file. Only stores with a generation allow Gc. *)
```

Comment:
It is possible to pattern match the payload in order to retrieve that information. Also, `` `Gc_disallowed`` is a bit out of concerns for the control file API